src/main/java/io/codiga/plugins/jetbrains/model/CodingAssistantVariableTransform.java

```diff
@@ -0,0 +1,81 @@
+package io.codiga.plugins.jetbrains.model;
+
+import com.intellij.codeInsight.completion.CompletionParameters;
+import com.intellij.openapi.vfs.VirtualFile;
+import static io.codiga.plugins.jetbrains.Constants.GET_FILENAME;
+import static io.codiga.plugins.jetbrains.Constants.GET_FILENAME_NO_EXT;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CodingAssistantVariableTransform {
+  private final CodingAssistantVariableContext codingAssistantVariableContext = new CodingAssistantVariableContext();
+
+  public String replaceVariables (String code,
+                                  CompletionParameters parameters,
+                                  VirtualFile file) {
+    List detectedVariables = detectVariablesInsideCode(code);
+    String processedCode = code;
+    System.out.println(detectedVariables);
+    if (detectedVariables.contains(GET_FILENAME)){
+      processedCode = transformGetFilename(processedCode,
+        parameters,
+        file
+      );
+    }
+    if (detectedVariables.contains(GET_FILENAME_NO_EXT)) {
+      processedCode = transformGetFilenameNoExt(processedCode,
+        parameters,
+        file
+      );
+    }
+    return processedCode;
+  }
+
+  private String transformGetFilename (String code,
+                                       CompletionParameters parameters,
+                                       VirtualFile virtualFile) {
+    if (parameters != null) {
+      final VirtualFile file = parameters.getOriginalFile().getVirtualFile();
+      final String filename = file.getName();
+      return code.replace(GET_FILENAME, filename);
+    }
+    if (virtualFile != null) {
+      final String filename = virtualFile.getName();
+      return code.replace(GET_FILENAME, filename);
+    }
+
+    return code;
+  }
+
+  private String transformGetFilenameNoExt (String code,
+                                            CompletionParameters parameters,
+                                            VirtualFile virtualFile) {
+    if (parameters != null) {
+      final VirtualFile file = parameters.getOriginalFile().getVirtualFile();
+      final String filename = file.getNameWithoutExtension();
+      return code.replace(GET_FILENAME_NO_EXT, filename);
+    }
+
+    if (virtualFile != null) {
+      final String filename = virtualFile.getName();
+      return code.replace(GET_FILENAME_NO_EXT, filename);
+    }
+
+    return code;
+  }
+
+  private List detectVariablesInsideCode(String code) {
+    String[] supportedVariables = codingAssistantVariableContext.getCodigaVariables();
+    List<String> foundVariables = new ArrayList<>();
+    for (int i = 0; i < supportedVariables.length; i++) {
+      boolean isFound = code.contains(supportedVariables[i]);
+      if (isFound) {
+        foundVariables.add(supportedVariables[i]);
+      }
+    }
+    return foundVariables;
+  }
```

Comment:
That piece of code can be re-written using the stream API.

Something like

```
Arrays.asList(SUPPORTED_VARIABLES).stream().filter(v => code.contains(v)).collect(Collectors.toList);
```

In that case, you probably do not need a function and can use this directly in the function.